### DISCUSSION TITLE
Added @clerk/nextjs and now I get 404s on my protected routes

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/next.js#66037] [Help]
### Summary

I am working on adding a `/admin/*` route to my website and decided to use `@clerk/nextjs` for this. However, I noticed that my `not-found.tsx` gets returned when developing locally when I am on a protected route.

This is what my route structure looks like:
<img width="324" alt="Screenshot 2024-05-21 at 8 43 38 AM" src="https://github.com/vercel/next.js/assets/21000200/545fe375-4fd8-4128-b76b-1a3ff0df04df">

However, when I go to `/admin/home` or `/admin/transfer-portal` I get my `not-found.tsx`

<img width="1725" alt="Screenshot 2024-05-21 at 8 44 37 AM" src="https://github.com/vercel/next.js/assets/21000200/cd9e004a-e424-4efc-870a-bdb7a4ccbbbc">

In my `/admin/home/page.tsx` I have

```tsx
const { userId } = auth()
const user = await currentUser()

console.log(user)
console.log('User ID:', userId)
```
 and both return data, so I know I am logged in.
 
 I am sure that I have something setup weird causing some potential race conditions.

### Additional information

```js
This is the `middleware.ts`

This is what my `app/layout.tsx` looks like


import './globals.css'

import { ClerkProvider } from '@clerk/nextjs'
import { GeistSans } from 'geist/font/sans'
import { GeistMono } from 'geist/font/mono'

import type { Viewport } from 'next'

export const viewport: Viewport = {
  themeColor: '#DC2727',
}

export default async function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <ClerkProvider>
      <html
        lang="en"
        suppressHydrationWarning
        className={`${GeistSans.variable} ${GeistMono.variable}`}
      >
        <body className="flex min-h-screen flex-col bg-background font-sans antialiased">
          {children}
        </body>
      </html>
    </ClerkProvider>
  )
}


```ts
import { clerkMiddleware, createRouteMatcher } from '@clerk/nextjs/server'

const isProtectedRoute = createRouteMatcher(['/admin(.*)'])

export default clerkMiddleware((auth, req) => {
  if (isProtectedRoute(req)) auth().protect()
})

export const config = {
  matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
}
```
```


### Example

_No response_

**Comment by colinclerk** [comment]
Colin from Clerk here. We've seen this on latest as well and are investigating if it's a Next.js issue or a Clerk issue.

**Comment by colinclerk** [comment]
@JamesSingleton have you seen this while navigating, or only after saving and the app does a fast refresh?

**Reply by JamesSingleton** [reply]
@colinclerk, So far I have only seen this while I have `http://localhost:3000/admin/home` open and I hit save in `app/admin/home/page.tsx`. Now I _did_ have `app/admin/page.tsx` instead of `app/admin/home/page.tsx` but kept seeing `[slug]/page.tsx` being recompiled every time I hit save so I moved away from that. I have not gotten this while going from lets say `/admin/home` to `/admin/transfer-portal` or `/admin/vote`.

**Reply by jojech** [reply]
I'm seeing this while navigating @colinclerk. Is there a solution for this?

**Reply by sheamcdevitt** [reply]
Still getting this error - is there a fix for this @colinclerk

**Comment by amanuel-s-tadesse** [comment]
this is a version is match it should be good after you upgrade your nextjs to latest version.
upgrading works for me

**Comment by hSwachchha** [comment]
Im also seeing the issue, apparently only the protected routes are getting "Page not found 404" errror, i have added custom middleware for each of the protected routes, and made them public in clerk middleware.js, as my project is small, im planning to use it that way, but i do want a permanent solution fast

**Comment by DanyloLyha** [comment]
Hello guys, in my case I'm facing 404 page after refresh on protected route. @colinclerk

**Reply by therealsharat** [reply]
Same, this is happening for my project as well

**Comment by shimorojune** [comment]
I am not using NextJS, but I had this same issue on my react app using clerk.

This was fixed for me by adding:
```json
{
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/index.html"
    }
  ]
}
```

to my `vercel.json` file. This essentially configures the server to serve `index.html` for all routes that don't match static files.

I am not sure if this will fix the issue in NextJS, but posted here regardless.
